pkg/api/steve/proxy/path.go

```diff
@@ -0,0 +1,38 @@
+package proxy
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// gvrFromPath tries to parse a URL path and extract GVR information from it.
+// Group and Version must be present. Resource may be empty for discovery client paths.
+func gvrFromPath(path string) (schema.GroupVersionResource, error) {
+	parts := strings.Split(path, "/")
+	if len(parts) < 4 {
+		return schema.GroupVersionResource{}, fmt.Errorf("path %s doesn't represent a valid GVR", path)
+	}
+	gvr := schema.GroupVersionResource{
+		Group:   parts[2],
+		Version: parts[3],
+	}
+	resourcePath := parts[4:]
+	switch len(resourcePath) {
+	case 1, 2:
+		// Non-namespaced resource.
+		gvr.Resource = resourcePath[0]
+	case 3:
+		// Sub-resource of either namespaced or non-namespaced resource.
+		prefix := resourcePath[0]
+		if prefix == "namespaces" {
```

Comment:
nit: would be good to register this as a const somewhere.

Reply:
This function is really self-contained, so this literal is already a constant, just unnamed. Not sure we'd get much by adding a constant outside the function.

Reply:
You can do a function-level const, so not sure how putting this outside of the function plays into this.

I see the benefit here as:
- Providing some guards against how this value is managed
- Declaring the arbitrary strings that you are using beforehand

Both of which I see beneficial here.

Reply:
You're talking about the "namespaces" string literal, right? I already have guards against how this value is used, since a literal is a constant. I don't think there is a need to declare this string beforehand, especially since I use it only once - else it becomes a sort of unnecessary stumble for the reader.